3MI0800270/2023-02-28/zad1.cpp

```diff
@@ -0,0 +1,80 @@
+#include <iostream>
+using namespace std;
+
+struct transition {
+    size_t to;
+    char c;
+};
+
+struct state {
+    transition transitions[4];
+    size_t num;
+};
+
+struct stateInfo {
+    state data[50];
+};
+
+bool hasTransition(const stateInfo& arg, size_t from, char c) {
+    size_t number = arg.data[from].num;
+    for (int j = 0;j < number;j++) {
+        if (arg.data[from].transitions[j].c == c) {
+            return true;
+        }
+    }
+}
+bool addTransition(stateInfo& arg, size_t from, char c, size_t to) {
+    size_t number = arg.data[from].num;
+    if (number >= 4) {
+        return false;
+    }
+    else {
+        arg.data[from].transitions[number].c = c;
+        arg.data[from].transitions[number].to = to;
+    }
+}
+int getTransition(const stateInfo& arg, size_t from, char c) {
+    size_t number = arg.data[from].num;
```

Comment:
You honestly don't need this as a separate variable. One thing to not here - you're using a mixture of `size_t` and `int` across the code - this is far from ideal imho (`int i` vs `size_t number`). My 5 cents to that are - don't overdo do things, using a plain and basic `int` is more than fine.